consensus/state.go

```diff
@@ -556,6 +556,7 @@ func (cs *ConsensusState) receiveRoutine(maxSteps int) {
 	defer func() {
 		if r := recover(); r != nil {
 			cs.Logger.Error("CONSENSUS FAILURE!!!", "err", r, "stack", string(debug.Stack()))
+			go cs.receiveRoutine(0)
```

Comment:
We most probably shouldn't be running any further when there is some unexpected panic.  Some unknown error happened, and so we don't know if that will result in the validator signing an invalid thing.

Is this necessary?

I'd rather us crash... it might be worthwhile to explore a mechanism for manual resuming via some console or secure RPC system, but for now halting the chain upon unexpected consensus bugs sounds like the better option.

Reply:
Rather, how about we stop everything gracefully? (making sure things close properly if possible, esp autofile writes).

Reply:
Yep, I had the same thought yesterday. This largely depends on whenever we believe we can restore operations after consensus failure. I guess, for now, the safest route is to stop everything gracefully.